[Tan Guan Quan] iP

guanquann opened the pull request:
**GuanGuanBot**

> "Effective task management does not mean doing more things, but doing the right things. It's not about managing time, but managing priorities." _- A wise man_

GuanGuanBot is a chatbot designed for efficient task management to streamline your daily tasks and enhance productivity. It's,
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. Download the `.jar` file from [here](https://github.com/guanquann/ip/releases/download/A-Release/guanguan.jar).
2. Copy the jar file into an empty folder.
3. Open a command window in that folder.
4. Run the command `java -jar guanguan.jar` (i.e., run the command in the same folder as the jar file).
5. Add your tasks.
6. Let it manage your tasks for you 😉
7. PROFIT🤑!

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [x] Auto save
- [ ] Reminders (coming soon! _hopefully..._)

If you are a Java programmer, you can also use it to practice Java. Here's the method using Gradle:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

guanquann committed:
rename, greet, exit

guanquann committed:
echo

guanquann committed:
add, list

guanquann committed:
mark as done

guanquann committed:
todos, events, deadlines

guanquann committed:
automated text ui testing

guanquann committed:
handle errors

guanquann committed:
handle errors

guanquann committed:
delete task

guanquann committed:
save the tasks in hard disk

guanquann committed:
Merge branch 'branch-Level-7'

guanquann committed:
add date to tasks

guanquann committed:
Merge branch 'branch-Level-8'

guanquann committed:
Follow Java convention in Duke.java

guanquann committed:
Make the code more OOP

guanquann committed:
Merge branch 'A-MoreOOP'

guanquann committed:
Put classes into Duke package

guanquann committed:
Merge branch 'branch-A-Packages'

guanquann committed:
Merge branch 'add-gradle-support' of https://github.com/guanquann/ip into branch-A-Gradle

guanquann committed:
Update build.gradle

guanquann committed:
Merge branch 'branch-A-Gradle'

guanquann committed:
Add unit test for Deadline, Event, Parser, TaskList, Todo

guanquann committed:
Make task abstract class

guanquann committed:
Add JavaDoc comments

guanquann committed:
Tweak Event, Deadline, Storage to comply with coding standard

guanquann committed:
Find a task by searching for a keyword

guanquann committed:
Merge tag 'A-JavaDoc'

Add JavaDoc comments

guanquann committed:
Merge tag 'A-CodingStandard'

Tweak code to comply with a coding standard

guanquann committed:
Resolve merge conflict in TaskList

guanquann committed:
Update bot and filenames to GuanGuan

guanquann committed:
Add switch stastements to Parser and Storage

guanquann committed:
Add GUI to chatbot using JavaFX

guanquann committed:
Add GUI to chatbot using JavaFX

guanquann committed:
Merge branch 'branch-Level-10'

guanquann committed:
Add checkstyle to detect coding style violations

guanquann committed:
Merge branch 'branch-A-CheckStyle'

guanquann committed:
Add abstract method to Task, fix GUI bye animation

azumieflare commented on src/main/java/guanguan/GuanGuan.java:
```diff
@@ -0,0 +1,76 @@
+ * Main class of chatbot.
+ */
+public class GuanGuan {
+    private final Storage storage;
+    private final TaskList items;
+    private final Ui ui;
```
final variables should be in all caps

azumieflare commented on src/main/java/guanguan/Gui.java:
```diff
@@ -0,0 +1,92 @@
+        return out;
+    }
+
+    @Override
```
Good note that javadocs is not required as it performs a similar job as the parent method that it is overriding.

azumieflare commented on src/main/java/guanguan/Gui.java:
```diff
@@ -0,0 +1,92 @@
+     * Get accumulated output from the UI.
+     * @return Text output accumulated so far.
+     */
+    public String getTextOutput() {
```
Maybe could add @override here? Or could remove ```getTextOutput()``` from ```Ui.java``` as it seems to do nothing

azumieflare commented on src/main/java/guanguan/Ui.java:
```diff
@@ -0,0 +1,113 @@
+    /**
+     * Prints empty line message
+     */
+    public void emptyLine() {
```
This seems like a redundant method? Could just do ```System.out.println();``` instead of making a method for this?

azumieflare reviewed (commented):
Overall looks good!

guanquann committed:
Use assert features in GuanGuan

guanquann commented:
@azumieflare thank you for your comment!

guanquann committed:
Improve code quality using the guidelines in the website

guanquann committed:
Merge pull request #1 from guanquann/branch-A-Assertions

Use assert features in GuanGuan

guanquann committed:
Merge pull request #2 from guanquann/branch-A-CodeQuality

Improve code quality using the guidelines in the website

guanquann committed:
Use GitHub Actions to set up Continuous Integration

Set up gradle.yml to perform checking during push to github

guanquann committed:
Update gradle.yml

guanquann committed:
Merge pull request #3 from guanquann/branch-A-CI

Branch a ci

guanquann committed:
Add mass operation to mark and delete tasks

Users can now perform tasks on multiple items at the same time.

This helps to reduce the time taken to run similar tasks.

guanquann committed:
Merge pull request #4 from guanquann/branch-C-MassOps

Add mass operation to mark and delete tasks

guanquann committed:
Add representative screenshot of the product

guanquann committed:
Add user guide

guanquann committed:
Update user guide with clearer instructions

guanquann committed:
Update jar name to guanguan

guanquann committed:
Improve code quality in GuanGuan and Task

guanquann committed:
Add command summary to docs

guanquann committed:
Fix table formatting in docs

guanquann committed:
Add title to GUI

guanquann committed:
Merge branch 'master' of https://github.com/guanquann/ip

guanquann committed:
Edit homepage readme, update Ui.png, change output task filename

guanquann committed:
Add setting up instructions in README.md

guanquann committed:
Update acknowledgement section
